numba/parfors/parfor.py

```diff
@@ -3611,14 +3611,27 @@ def get_reduction_init(nodes):
     if acc_expr.fn == operator.iadd or acc_expr.fn == operator.isub:
         return 0, acc_expr.fn
     if (  acc_expr.fn == operator.imul
-       or acc_expr.fn == operator.itruediv
-       or acc_expr.fn == operator.ifloordiv ):
+       or acc_expr.fn == operator.itruediv ):
         return 1, acc_expr.fn
     return None, None
 
 def supported_reduction(x, func_ir):
     if x.op == 'inplace_binop' or x.op == 'binop':
-        return True
+        if x.fn == operator.ifloordiv or x.fn == operator.floordiv:
+            raise ValueError(("Parallel floordiv reductions are not supported. "
+                              "If all divisors are integers then a floordiv "
+                              "reduction can in some cases be parallelized as "
+                              "a multiply reduction followed by a floordiv of "
+                              "the resulting product."))
+        supps = [operator.iadd,
+                 operator.isub,
+                 operator.imul,
+                 operator.itruediv,
+                 operator.add,
+                 operator.sub,
+                 operator.mul,
+                 operator.truediv]
+        return x.fn in supps
```

Comment:
I think it would be good to update the `get_reduce_nodes` function to raise a `NumbaValueError` so that the exception handling is consistent. Would also be good to add a test for a reduction like `acc ^= True` or similar to ensure that this branch runs and is tested.

Reply:
I added the ^= test.  For the specific error to raise, there are numerous places in parfors.py where ValueError is raised and none where NumbaValueError is raised.  Do you want all those places changed in this PR or defer to a separate PR?

Reply:
Suggest fixing these couple of cases in this PR and then deferring the rest to a separate one. How does that sound?